internal/app/app.go

```diff
@@ -440,9 +448,14 @@ func (a App) cleanup() {
 	removeDirectory := func(directoryPath string) {
 		if _, err := os.Stat(directoryPath); !os.IsNotExist(err) {
 			if err = os.RemoveAll(directoryPath); err != nil {
-				exitCode = 2 // ENOENT
+				if a.softExit {
+					exitCode = 0
+					log.Infof("can't remove %s", directoryPath)
```

Comment:
you still have error information here, and could log it potentially, to ensure it is available in the logs.